plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataFileOperations.java

```diff
@@ -75,90 +83,220 @@ protected DistributedQueryRunner createQueryRunner()
+        assertThat(getOperations()).containsExactlyInAnyOrderElementsOf(expectedAccesses);
+    }
+
+    private static void resetCounts()
+            throws Exception
+    {
+        BEAN_SERVER.invoke(new ObjectName(BEAN_NAME), "reset", null, null);
+    }
+
+    private static Multiset<FileOperation> getOperations()
+            throws Exception
+    {
+        return ((Map<OperationContext, Integer>) BEAN_SERVER.getAttribute(new ObjectName(BEAN_NAME), "OperationCounts"))
+                .entrySet().stream()
+                .map(entry -> entry.getKey())
+                .map(FileOperation::new)
+                .collect(Collectors.toCollection(HashMultiset::create));
+    }
+
+    static class FileOperation
+    {
+        private final FileType fileType;
+        private final OperationType operationType;
+
+        public FileOperation(FileType fileType, OperationType operationType)
         {
-            this.allCounts = requireNonNull(allCounts, "allCounts is null");
+            this.fileType = requireNonNull(fileType, "fileType is null");
+            this.operationType = requireNonNull(operationType, "operationType is null");
         }
 
-        public OperationCounts forQueryId(String queryId)
+        public FileOperation(OperationContext operationContext)
         {
-            this.queryId = Optional.of(queryId);
-            return this;
+            this(fromFilePath(operationContext.getFilePath()), operationContext.getOperationType());
         }
 
-        public OperationCounts forPathContaining(String value)
+        @Override
+        public boolean equals(Object o)
         {
-            this.pathPredicate = path -> path.contains(value);
-            return this;
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            FileOperation that = (FileOperation) o;
+            return fileType == that.fileType && operationType == that.operationType;
         }
 
-        public OperationCounts forOperation(TrackingFileIoProvider.OperationType type)
+        @Override
+        public int hashCode()
         {
-            this.type = Optional.of(type);
-            return this;
+            return Objects.hash(fileType, operationType);
         }
 
-        public Map<OperationContext, Integer> get()
+        @Override
+        public String toString()
         {
-            return allCounts.entrySet().stream()
-                    .filter(entry -> queryId.map(id -> entry.getKey().getQueryId().equals(id)).orElse(true))
-                    .filter(entry -> pathPredicate.test(entry.getKey().getFilePath()))
-                    .filter(entry -> type.map(value -> value == entry.getKey().getOperationType()).orElse(true))
-                    .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
+            final StringBuilder sb = new StringBuilder("FileOperation{");
```

Comment:
we do not mark variables/parameters as `final`

for toString, `toStringHelper` is preferred